internal/infrastructure/kubernetes/infra.go

```diff
@@ -32,11 +38,22 @@ type Resources struct {
 
 // NewInfra returns a new Infra.
 func NewInfra(cli client.Client) *Infra {
-	return &Infra{
+	infra := &Infra{
 		mu:        sync.Mutex{},
 		Client:    cli,
 		Resources: newResources(),
 	}
+
+	// Set the namespace used for the managed infra.
+	ns, found := os.LookupEnv("ENVOY_GATEWAY_NAMESPACE")
```

Comment:
can use some tool like this https://github.com/istio/pkg/blob/8b4fffa5f37157b1dc5e3aa03b0f88c2a92f55e5/env/var.go#L128

Reply:
its eventually doing the same thing :) https://github.com/istio/pkg/blob/8b4fffa5f37157b1dc5e3aa03b0f88c2a92f55e5/env/var.go#L230
can you clarify how we gain by adopting something similar now ?

Reply:
if there's more than one env vars to handle, the code will smell not good.

Reply:
yah agree, we should def find a way to reuse logic once we have more such env vars

Reply:
> if there's more than one env vars to handle, the code will smell not good.

@zirain feel free to create an issue and add it to the backlog milestone.